src/statement/mod.rs

```diff
@@ -172,6 +183,55 @@ impl Raii<ffi::Stmt> {
         }
     }
 
+    fn describe_col(&self, idx: u16) -> Return<ColumnDescriptor> {
+        let mut name_buffer = vec![0u8; 512];
+        let mut name_length: ffi::SQLSMALLINT = 0;
+        let mut data_type: ffi::SqlDataType = ffi::SqlDataType::SQL_UNKNOWN_TYPE;
+        let mut column_size: ffi::SQLULEN = 0;
+        let mut decimal_digits: ffi::SQLSMALLINT = 0;
+        let mut nullable: ffi::SQLSMALLINT = 0;
+        unsafe {
+            match ffi::SQLDescribeCol(self.handle(),
+                                      idx,
+                                      name_buffer.as_mut_ptr(),
+                                      name_buffer.len() as ffi::SQLSMALLINT,
+                                      &mut name_length as *mut ffi::SQLSMALLINT,
+                                      &mut data_type as *mut ffi::SqlDataType,
+                                      &mut column_size as *mut ffi::SQLULEN,
+                                      &mut decimal_digits as *mut ffi::SQLSMALLINT,
+                                      &mut nullable as *mut ffi::SQLSMALLINT
+            ) {
+                SQL_SUCCESS => Return::Success(ColumnDescriptor {
+                    name: ::std::str::from_utf8(&name_buffer[..(name_length as usize)]).unwrap().to_owned(),
+                    data_type: data_type,
+                    column_size: if column_size == 0 { None } else { Some(column_size) },
+                    decimal_digits: if decimal_digits == 0 { None } else { Some(decimal_digits as u16) },
+                    nullable: match nullable {
+                        2 /*ffi::SQL_NULLABLE_UNKNOWN*/ => None,
+                        1 /*ffi::SQL_NULLABLE*/ => Some(true),
+                        0 /*ffi::SQL_NO_NULLS*/ => Some(false),
+                        n =>  panic!("SQLDescribeCol returned unexpected nullable value: {:?}", n)
+                    }
+                }),
+                SQL_SUCCESS_WITH_INFO => Return::SuccessWithInfo(ColumnDescriptor {
+                    name: ::std::str::from_utf8(&name_buffer[..(name_length as usize)]).unwrap().to_owned(),
```

Comment:
`name_buffer` is `Vec` is already allocated on the heap. Therefore creating a `str` reference to it and reallocating it is more work than is actually needed here. We can either replace this function call with `std::string::String::from_utf8` which directly takes a `Vec` and returns a `String` instead of `str`, or (even better) keep this line as it is and change the type of `name_buffer` to `[u8, 512]`. I'd prefer the second solution since it defers the dynamic allocation until we know the length of the `String`